modules/openapi-generator/src/main/resources/typescript-nestjs/api.service.mustache

```diff
@@ -144,17 +147,16 @@ export class {{classname}} {
 {{/isBasicBasic}}
 {{#isBasicBearer}}
         if (typeof this.configuration.accessToken === 'function') {
-            headers['Authorization'] = `Bearer ${this.configuration.accessToken()}`;
+            accessTokenObservable = from(Promise.resolve(this.configuration.accessToken()));
```

Comment:
Encapsulation with `Promise.resolve(...)` allows `from` to be used irrespective of whether the returned value is a Promise or not